src/AutoRest.CSharp.V3/Generation/Writers/CodeWriter.cs

```diff
@@ -263,6 +264,7 @@ public CodeWriter Literal(object? o)
                 double d => SyntaxFactory.Literal(d).ToString(),
                 float f => SyntaxFactory.Literal(f).ToString(),
                 bool b => b ? "true" : "false",
+                EnumTypeValue enumTypeValue => enumTypeValue.Declaration.Type.Name + "." + enumTypeValue.Declaration.Name,
```

Comment:
Every type usage has to go through AppendType method so usings are correctly added.

Reply:
Can we add this logic to WriteConstant instead. Enum values are not really literals.